hadoop-hdds/rocksdb-checkpoint-differ/src/main/java/org/apache/ozone/rocksdiff/RocksDBCheckpointDiffer.java

```diff
@@ -939,6 +980,290 @@ private void populateCompactionDAG(List<String> inputFiles,
 
   }
 
+  /**
+   * This is the task definition which is run periodically by the service
+   * executor at fixed delay.
+   * It looks for snapshots in compaction DAG which are older than the allowed
+   * time to be in compaction DAG and removes them from the DAG.
+   */
+  public void pruneOlderSnapshotsWithCompactionHistory() {
+    List<Path> olderSnapshotsLogFilePaths =
+        getOlderSnapshotsCompactionLogFilePaths();
+
+    List<String> lastCompactionSstFiles =
+        getLastCompactionSstFiles(olderSnapshotsLogFilePaths);
+
+    Set<String> sstFileNodesRemoved =
+        pruneSnapshotFileNodesFromDag(lastCompactionSstFiles);
+    removeSstFile(sstFileNodesRemoved);
+    deleteOlderSnapshotsCompactionFiles(olderSnapshotsLogFilePaths);
+  }
+
+  /**
+   * Deletes the SST file from the backup directory if exists.
+   */
+  private void removeSstFile(Set<String> sstFileNodes) {
+    for (String sstFileNode: sstFileNodes) {
+      File file = new File(sstBackupDir + sstFileNode + SST_FILE_EXTENSION);
+      try {
+        Files.deleteIfExists(file.toPath());
```

Comment:
Note: `deleteIfExists` is fine. But in which case would SST not be found when it is being deleted here. Is it worth throwing a warning at least if the file doesn't exist?

Reply:
Case is when compacted SST files get deleted by the SST file clean up thread.
e.g. 
A, B, C get compacted to D, E
D, E, F, G get compacted to H, I

We don't need D and E to generate diff and can be deleted by `SST file clean up thread`. Tho these nodes will be in DAG, we can still remove them.